File: app/src/main/java/com/blacklenspub/starwarsapp/StarWarsApi.java
package com.blacklenspub.starwarsapp;

import retrofit2.Call;
import retrofit2.http.GET;
import retrofit2.http.Path;

public interface StarWarsApi {
    @GET("films/")
    Call<FilmResponse> getAllFilms();

    @GET("films/{id}/")
    Call<Film> getFilm(@Path("id") long id);
}

File: app/src/main/kotlin/com/blacklenspub/starwarsapp/StarWarsApi.kt
package com.blacklenspub.starwarsapp

import retrofit2.Call
import retrofit2.http.GET
import retrofit2.http.Path

interface StarWarsApi {

    @GET("films/")
    fun getAllFilms(): Call<FilmResponse>

    @GET("films/{id}/")
    fun getFilm(@Path("id") id: Long): Call<Film>
}
